webproject/src/main/java/com/google/sps/dataManagers/UserRoomManager.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.dataManagers;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.ServletException;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.Query;
+import com.google.sps.data.UserRoomProto.UserRoom;
+import com.google.sps.util.FirebaseUtil;
+import java.util.HashMap;
+import java.util.Map;
+
+@Singleton
+public class UserRoomManager {
+    private FirebaseUtil firebaseUtil;
+
+    @Inject
+    public UserRoomManager(FirebaseUtil firebaseUtil) {
+        this.firebaseUtil = firebaseUtil;
+    }
+
+    public UserRoom addUserRoom(String userEmail, String roomId) throws ServletException {
+        if (!isRoomIdValid(roomId)) {
+            throw new ServletException("Invalid roomId");
+        }
+     
+        String userEmailRoom = userEmail + "_" + roomId;
+        Map<String, String> userRoomMapping = new HashMap<>();
+        userRoomMapping.put("userEmail", userEmail);
+        userRoomMapping.put("roomId", roomId);
+        userRoomMapping.put("userEmailRoom", userEmailRoom);
+
+        firebaseUtil.getUserRoomReference().push()
+                .setValue(userRoomMapping, (databaseError, databaseReference) -> {
+                    if (databaseError != null) {
+                        try {
+                            throw new ServletException("Invalid roomId");
+                        } catch (ServletException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                });
+        
+        UserRoom userRoom =
+                UserRoom.newBuilder().setUserEmail(userEmail).setRoomId(roomId).setUserEmailRoom(userEmailRoom).build();
+        return userRoom;
+    }
+
+    public UserRoom getUserRoom(String userEmail, String roomId) throws ServletException {
+        if (!isRoomIdValid(roomId)) {
+            throw new ServletException("Invalid roomId");
+        }
+
+        String userEmailRoom = userEmail + "_" + roomId;
+        UserRoom userRoom =
+                UserRoom.newBuilder().setUserEmail(userEmail).setRoomId(roomId).setUserEmailRoom(userEmailRoom).build();
+        
+        Query query = firebaseUtil.getUserRoomReference().orderByChild("userEmailRoom");
+        
+        Optional<DataSnapshot> dataSnapshot = firebaseUtil.getQuerySnapshot(query, userEmailRoom);
+        if (dataSnapshot.isPresent()) {
```

Comment:
its better to return early, lets change this to !datasnapshot.ispresent() and return null

Reply:
okay!